helix-core/src/main/java/org/apache/helix/cloud/event/helix/DefaultCloudEventCallbackImpl.java

```diff
@@ -90,10 +90,9 @@ public void exitMaintenanceMode(HelixManager manager, Object eventInfo) {
     // Check if there is any disabled live instance that was disabled due to cloud event,
     // if none left, exit maintenance mode
     HelixDataAccessor accessor = manager.getHelixDataAccessor();
-    if (instances.stream().noneMatch(instance ->
-        InstanceValidationUtil.getInstanceHelixDisabledType(accessor, instance)
-            .equals(InstanceConstants.InstanceDisabledType.CLOUD_EVENT.name())
-            && InstanceValidationUtil.isAlive(accessor, instance))) {
+    if (instances.stream().noneMatch(instance -> HelixEventHandlingUtil
+        .IsInstanceDisabledForCloudEvent(manager.getClusterName(), instance, accessor.getBaseDataAccessor())
+        && InstanceValidationUtil.isAlive(accessor, instance))) {
```

Comment:
With current implementation of `IsInstanceDisabledForCloudEvent`, it's always returning true, and we won't exit maintenance mode at all. Is this an issue?